harvester_e2e_tests/apis/test_images.py

```diff
@@ -85,11 +86,31 @@ def test_update(self, api_client, unique_name):
                 "test-label": "42"
             },
             "annotations": {
-                "test-annotation": "dummy"
-            }
+                "test-annotation": "dummy",
+                "field.cattle.io/description": 'test description'
+            },
+
         }
 
         code, data = api_client.images.update(unique_name, dict(metadata=updates))
+
+        assert 200 == code, (f"Failed to update image with error: {code}, {data}")
+
+        unexpected = list()
+        for field, pairs in updates.items():
+            for k, val in pairs.items():
+                if data['metadata'][field].get(k) != val:
+                    unexpected.append((field, k, val, data['metadata'][field].get(k)))
+
+        assert not unexpected, (
+            "\n".join(f"Update {f} failed, set key {k} as {v} but got {n}"
+                      for f, k, v, n in unexpected)
+        )
+
+    def test_update_by_input(self, api_client, unique_name, updates):
```

Comment:
fixture `updates` is missing, when modifying old test cases, we should always test it.

for the two test cases, you might need check it again, update then test it.

Reply:
The `updates` used in  `test_update_by_input` act as parameter called from integration/test_images.py 
Moved the `test_update_by_input`, `test_upload_file` and `test_get_export` from apis/test_images.py to integration/test_images.py to isolate the test and wrap them in `TestBackendImages` class